lib/galaxy/tools/errors.py

```diff
@@ -91,30 +177,37 @@ def create_report( self, user, email='', message='', **kwd ):
-        self.report = string.Template( error_report_template ) \
-            .safe_substitute( host=host,
-                              dataset_id_encoded=self.app.security.encode_id( hda.dataset_id ),
-                              dataset_id=hda.dataset_id,
-                              history_id_encoded=history_id_encoded,
-                              history_id=hda.history_id,
-                              hda_id_encoded=hda_id_encoded,
-                              hid=hda.hid,
-                              history_item_name=hda.get_display_name(),
-                              history_view_link=history_view_link,
-                              hda_show_params_link=hda_show_params_link,
-                              job_id_encoded=self.app.security.encode_id( job.id ),
-                              job_id=job.id,
-                              tool_version=job.tool_version,
-                              job_tool_id=job.tool_id,
-                              job_tool_version=hda.tool_version,
-                              job_runner_external_id=job.job_runner_external_id,
-                              job_command_line=job.command_line,
-                              job_stderr=util.unicodify( job.stderr ),
-                              job_stdout=util.unicodify( job.stdout ),
-                              job_info=util.unicodify( job.info ),
-                              job_traceback=util.unicodify( job.traceback ),
-                              email_str=email_str,
-                              message=util.unicodify( message ) )
+
+        report_variables = dict(
+            host=host,
+            dataset_id_encoded=self.app.security.encode_id( hda.dataset_id ),
+            dataset_id=hda.dataset_id,
+            history_id_encoded=history_id_encoded,
+            history_id=hda.history_id,
+            hda_id_encoded=hda_id_encoded,
+            hid=hda.hid,
+            history_item_name=hda.get_display_name(),
+            history_view_link=history_view_link,
+            hda_show_params_link=hda_show_params_link,
+            job_id_encoded=self.app.security.encode_id( job.id ),
+            job_id=job.id,
+            tool_version=job.tool_version,
+            job_tool_id=job.tool_id,
+            job_tool_version=hda.tool_version,
+            job_runner_external_id=job.job_runner_external_id,
+            job_command_line=job.command_line,
+            job_stderr=util.unicodify( job.stderr ),
+            job_stdout=util.unicodify( job.stdout ),
+            job_info=util.unicodify( job.info ),
+            job_traceback=util.unicodify( job.traceback ),
+            email_str=email_str,
+            message=util.unicodify( message )
+        )
+
+        self.report = string.Template( error_report_template ).safe_substitute( report_variables )
+        # Escape the message for use in the HTML report
+        report_variables['message'] = cgi.escape(report_variables['message'])
```

Comment:
I suspect that `message` is not the only variable which can be considered (at least partially) under the user control: history_item_name, job_command_line, job_stderr, job_stdout, job_traceback